app/api/auth/authMiddleware.ts

```diff
@@ -0,0 +1,23 @@
+import { Request, Response, NextFunction } from 'express';
+
+declare global {
+  namespace Express {
+    export interface Request {
+      user: import('../users/usersModel').User;
```

Comment:
This differs from konzz's PR.
Using the import statement avoids having to disable the rule for not used vars and it is the preferred way of importing external types to a global definition based on several articles.